addons/survey/static/src/js/survey_renderers.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module */
+
+import Dialog from 'web.Dialog';
+import KanbanRenderer from 'web.KanbanRenderer';
+import ListRenderer from 'web.ListRenderer';
+import { qweb as QWeb, _t } from 'web.core';
+
+const SurveySampleModalMixin = {
+    _onOpenModal: function () {
+        const $content = $(QWeb.render('survey.survey_sample_modal_body'));
+        $content.find('.o_survey_sample_tile').each((_index, tile) => {
+            const $tile = $(tile);
+            $tile.on('click', () => {
+                this._rpc({
+                    model: 'survey.survey',
+                    method: $tile.data('action'),
+                }).then(action => {
+                    this.do_action(action);
+                });
```

Comment:
Can't we await here to get less indentation?

```js
const surveySampleAction = await this._rpc({
    model: 'survey.survey',
    method: $tile.data('action'),
})

this.do_action(surveySampleAction);
```

(You might need to make the whole handler method async).

Reply:
Hmmm I don't really agree.

Currently, if the server does not answer, your feature does not work.
So is a raised error really that much worse?

I still vote for the await :smile: